Recipes.Tests/WebApiTelemetryTests.cs

```diff
@@ -138,16 +140,36 @@ public void TearDown()
             }
 
             telemetryEvents.Single()
-                           .IsIncomingRequest()
-                           .Should()
-                           .BeTrue();
-            telemetryEvents.Single()
                            .CallerIpAddress()
                            .Should()
                            .Be("123.123.123.123");
         }
 
         [Test]
+        public async Task Telemetry_events_based_on_HTTP_request_are_marked_as_incoming()
+        {
+            HttpResponseMessage response = null;
+
+            using (Log.With<Telemetry>(t => t.WithPropertiesBasedOn(response)).Enter(() => { }))
+            {

```

Comment:
I think the test name is still overly broad. It implies all requests based on responses are marked as incoming. The next test has a "when" and I think this one needs one as well.